ui/src/overridden/backoffice/Importer/Importer.js

```diff
@@ -0,0 +1,253 @@
+import React, { Component } from 'react';
+import {
+  Container,
+  Header,
+  Segment,
+  Button,
+  Form,
+  Modal,
+  Icon,
+  Label,
+  Message,
+} from 'semantic-ui-react';
+import { invenioConfig } from '@inveniosoftware/react-invenio-app-ils';
+import { ImportedDocuments } from './ImportedDocuments';
+import _isEmpty from 'lodash/isEmpty';
+import _get from 'lodash/get';
+import { http } from '@inveniosoftware/react-invenio-app-ils';
+
+export default class Importer extends Component {
+  constructor(props) {
+    super(props);
+    this.filesRef = React.createRef();
+    this.state = {
+      provider: '',
+      mode: '',
+      file: null,
+      openModal: false,
+      postDone: false,
+      providerMissing: false,
+      modeMissing: false,
+      fileMissing: false,
+      error: {},
+      taskId: null,
+    };
+  }
+
+  handleChange = (e, { name, value }) =>
+    this.setState({
+      [name]: value,
+      providerMissing: false,
+      modeMissing: false,
+      fileMissing: false,
+    });
+
+  postData = async formData => {
+    const importerURL = '/importer';
+    const headers = {
+      headers: {
+        'Content-Type': 'multipart/form-data',
+      },
+    };
+    try {
+      const response = await http.post(`${importerURL}`, formData, headers);
+      this.setState({
+        provider: '',
+        mode: '',
+        file: null,
+        postDone: true,
+        taskId: response.data.id,
+        error: {},
+      });
+    } catch (error) {
+      this.setState({
+        error: error,
+      });
+    }
+  };
+
+  handleSubmit = async () => {
+    const { provider, mode, file } = this.state;
+    if (!_isEmpty(provider) && !_isEmpty(mode) && file) {
+      var formData = new FormData();
```

Comment:
we are using a React upload component to upload files for eitems, why not re-using that one and get rid of custom code?

Reply:
Hmm, if you mean this component I did not use it because it seems to specific for the eitems, it needs an eitem as a prop, it launches the call on selecting the file, while we only want to store it and send it along with the rest of the data from the form. Maybe I can create a more generic component re-usable by both use cases